lib/objects/objectsInMemFileDB.js

```diff
@@ -0,0 +1,2301 @@
+            if (typeof callback === 'function') {
+                setImmediate(() => callback(null, processed, id));
+            }
+        } catch (e) {
+            if (typeof callback === 'function') {
+                setImmediate(() => callback(e.message));
+            }
+        }
+    }
+    chmodFile(id, name, options, callback) {
+        if (typeof options === 'function') {
+            callback = options;
+            options = null;
+        }
+        options = options || {};
+        options.acl = null;
+
+        const _path = utils.sanitizePath(id, name, callback);
+        if (!_path) return;
+        id = _path.id;
+        name = _path.name;
+
+        if (typeof options !== 'object') {
+            options = {mode: options};
+        }
+
+        if (options.mode === undefined) {
+            this.log.error(this.namespace + ' mode is not defined');
+            if (typeof callback === 'function') callback('invalid parameter');
+            return;
+        } else if (typeof options.mode === 'string') {
+            options.mode = parseInt(options.mode, 16);
+        }
+
+        this.checkFileRights(id, null, options, utils.CONSTS.ACCESS_WRITE, (err, options) => {
+            if (err) {
+                if (typeof callback === 'function') callback(err);
+            } else {
+                if (!options.acl.file.write) {
+                    if (typeof callback === 'function') callback(utils.ERRORS.ERROR_PERMISSION);
+                } else {
+                    return this._chmodFile(id, name, options, callback);
+                }
+            }
+        });
+    }
+
+    _enableFileCache(enabled, _options, callback) {
+        if (this.settings.connection.noFileCache !== enabled) {
+            this.settings.connection.noFileCache = !!enabled;
+            if (!this.settings.connection.noFileCache) {
+                // clear cache
+                this.files = {};
+            }
+        }
+        if (typeof callback === 'function') {
+            setImmediate(() => callback(null, this.settings.connection.noFileCache));
+        }
+    }
+    enableFileCache(enabled, options, callback) {
+        if (typeof options === 'function') {
+            callback = options;
+            options = null;
+        }
+
+        if (options && options.acl) {
+            options.acl = null;
+        }
+
+        utils.checkObjectRights(this, null, null, options, utils.CONSTS.ACCESS_WRITE, (err, options) => {
+            if (err) {
+                if (typeof callback === 'function') callback(err);
+            } else {
+                return this._enableFileCache(enabled, options, callback);
+            }
+        });
+    }
+
+    // -------------- OBJECT FUNCTIONS -------------------------------------------
+    clone(obj) {
+        if (obj === null || obj === undefined || typeof obj !== 'object') {
+            return obj;
+        }
+
+        const temp = obj.constructor(); // changed
```

Comment:
can we do that without `new`?

Reply:
I have no idea .. I only know that it works that way all the time ...